app/models/container_image.rb

```diff
@@ -11,6 +11,7 @@ class ContainerImage < ApplicationRecord
   DOCKER_PULLABLE_PREFIX = "docker-pullable://".freeze
   DOCKER_PREFIXES = [DOCKER_IMAGE_PREFIX, DOCKER_PULLABLE_PREFIX].freeze
 
+  belongs_to :parent, :polymorphic => true
```

Comment:
To me it sounds like what you are trying to get is genealogy, in which case, containers should use the genealogy aspects that we already have for VMs (i.e. we have a relationship tree that show which instance came from which image, and then which image came from another image all the way back)  @simon3z I'm not sure if this is already being done elsewhere in the containers code, but if so, we should make it consistent.